src/main.ts

```diff
@@ -250,9 +256,12 @@ async function startServer(server: http.Server | https.Server) {
 
   const config = await getViteConfig();
   const isUsingViteResolvedConfig = Object.entries(config).length > 3;
+  const configDefaults = isUsingViteResolvedConfig
+    ? { root: config.root }
+    : config;
```

Comment:
Why do you want to override `root` setting from the `vite.config`? I don't want to mix those two thing, if there is some use-case that I don't know of, let me know and we will figure something out.

Reply:
Ah yes. I arrived at this solution because vite still wasn't finding the config when starting the server.
Looking at the source, I saw it was calling `resolveConfig` again: https://github.com/vitejs/vite/blob/main/packages/vite/src/node/server/index.ts#L389

So if we don't specify the root here, it will again default to `process.cwd()`. I'm just having it pass through whatever root was previously resolved by vite here: https://github.com/szymmis/vite-express/blob/f62567216dc43a0ac7c9127aed55339c409984f1/src/main.ts#L89

If viteConfigFile is not specified, than it will be vite's default (`process.cwd()`). And if vite is not being used to resolve the config, your manual config object will be passed, same as before

Reply:
Okay, so maybe we could just pass `configFile` into the `mergeConfig` function itself. That way the intent what we try to do is much clearer, what do you think?

Reply:
Indeed, that is much clearer! (I confirmed it still works in my monorepo)